app/src/src_nonfreenet/org/breezyweather/sources/metoffice/json/MetOfficeForecast.kt

```diff
@@ -0,0 +1,52 @@
+/**
+ * This file is part of Breezy Weather.
+ *
+ * Breezy Weather is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, version 3 of the License.
+ *
+ * Breezy Weather is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Breezy Weather. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.breezyweather.sources.metoffice.json
+
+import kotlinx.serialization.Serializable
+import org.breezyweather.common.serializer.DateUtcSerializer
+import java.util.Date
+
+@Serializable
+data class MetOfficeGeometry(
+    val type: String?,
+    val coordinates: List<Double>?
+)
+
+@Serializable
+data class MetOfficeLocation(
+    val name: String?
```

Comment:
Can this be used for reverse geocoding? Better than just "Current location"

Reply:
The API can include the location name in the response data so I guess so. Am I correct that the only way to implement this within Breezy is as a separate API call? Strictly speaking it's not necessary as the location name is just returned along with the forecast data itself but not sure if there is a way to use the location data from within `requestWeather`.

Reply:
Yes, you will need to implement a function that will call the same endpoint since reverse geocoding and weather are at the same endpoint for this source.

Unless badly designed server side, we have a cache support that is able to not make it actually call twice the server ;)